#! /bin/sh -e

# DP: fix bashism

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
diff -urNd isdnutils-3.9.20060704.orig/eurofile/scripts/eftd.sh.in isdnutils-3.9.20060704/eurofile/scripts/eftd.sh.in
--- isdnutils-3.9.20060704.orig/eurofile/scripts/eftd.sh.in	2008-08-24 15:48:48.000000000 +0100
+++ isdnutils-3.9.20060704/eurofile/scripts/eftd.sh.in	2008-08-24 15:50:36.000000000 +0100
@@ -155,7 +155,7 @@
     fi
 
     if test -z $EFT_SERVER_MSN; then
-      echo $'\a'"Error: You must define an MSN/EAZ in @I4LCONFDIR@/eft.conf"
+      printf "\aError: You must define an MSN/EAZ in @I4LCONFDIR@/eft.conf\n"
       exit 1
     fi
 
